include/deal.II/base/utilities.h

```diff
@@ -1030,27 +1030,21 @@ namespace Utilities
 {
   template <int N, typename T>
   inline T
-  fixed_power(const T n)
+  fixed_power(const T x)
   {
-    Assert(N >= 0, ExcNotImplemented());
-    switch (N)
-      {
-        case 0:
-          return dealii::internal::NumberType<T>::value(1);
-        case 1:
-          return n;
-        case 2:
-          return n * n;
-        case 3:
-          return n * n * n;
-        case 4:
-          return n * n * n * n;
-        default:
-          T result = n;
-          for (int d = 1; d < N; ++d)
-            result *= n;
-          return result;
-      }
+    Assert(
+      !std::is_integral<T>::value || (N >= 0),
+      ExcMessage(
+        "The non-type template parameter N must be a non-negative integer for integral type T"));
+
+    if (N == 0)
+      return T(1.);
+    else if (N < 0)
+      return T(1.) / fixed_power<-N>(x);
```

Comment:
For reference, you could have stuck with the `static_assert` if you had made sure that the function here is never called with a negative argument. This line is executed if `N<0`, so what you're really calling here is `fixed_power<abs(N)>`, which you can write as `fixed_power<(N>=0 ? N : -N>`. This way, no code paths explicitly call `fixed_power` with a negative argument, and you can revert to the `static_assert`.

If you're up to it, feel free to propose the change as a follow-up (with a comment explaining the funny conditional); if not, let me know and I'll take care of it!

Reply:
Good suggestion! https://godbolt.org/z/qZHi5d confirms that this works.